client/my-sites/importer/section-import.jsx

```diff
@@ -130,98 +121,81 @@ class SectionImport extends Component {
 	} );
 
 	componentDidMount() {
-		ImporterStore.on( 'change', this.updateState );
 		this.updateFromAPI();
+		if ( this.props.isImporterStatusHydrated ) {
+			this.onceAutoStartImport();
+		}
 	}
 
 	componentDidUpdate() {
-		const { site } = this.props;
-
-		if ( ! site.ID ) {
-			return;
+		if ( this.props.isImporterStatusHydrated ) {
+			this.onceAutoStartImport();
 		}
 
-		this.onceAutoStartImport();
 		this.handleStateChanges();
 	}
 
-	componentWillUnmount() {
-		ImporterStore.off( 'change', this.updateState );
-	}
-
 	/**
 	 * Renders each enabled importer at the provided `state`
 	 *
-	 * @param {object} site Data for the currently active site
-	 * @param {string} siteTitle The site's title
-	 * @param {string} state The state constant for the importer components
+	 * @param {string} importerState The state constant for the importer components
 	 * @returns {Array} A list of react elements for each enabled importer
 	 */
-	renderIdleImporters( site, siteTitle, state ) {
-		const {
-			options: { is_wpcom_atomic: isAtomic },
-		} = site;
+	renderIdleImporters( importerState ) {
+		const { site, siteTitle } = this.props;
+		let importers = getImporters();
 
-		const importerElementsAll = getImporters();
-
-		/**
-		 * Filter out all importers except the WordPress ones for Atomic sites.
-		 */
-		const importerElementsFiltered = isAtomic
-			? importerElementsAll.filter( ( importer ) => importer.engine === 'wordpress' )
-			: importerElementsAll;
+		// Filter out all importers except the WordPress ones for Atomic sites.
+		if ( site.options.is_wpcom_atomic ) {
+			importers = importers.filter( ( importer ) => importer.engine === 'wordpress' );
+		}
 
-		const importerElements = importerElementsFiltered.map( ( importer ) => {
-			const { engine } = importer;
+		const importerElements = importers.map( ( { engine } ) => {
 			const ImporterComponent = importerComponents[ engine ];
 
 			if ( ! ImporterComponent ) {
-				return;
+				return null;
 			}
 
+			const importerStatus = {
+				importerState,
+				type: getImporterTypeForEngine( engine ),
+			};
+
 			return (
 				<ImporterComponent
 					key={ engine }
 					site={ site }
 					siteTitle={ siteTitle }
-					importerStatus={ {
-						importerState: state,
-						type: getImporterTypeForEngine( engine ),
-					} }
+					importerStatus={ importerStatus }
 				/>
 			);
 		} );
 
 		// add the 'other importers' card to the end of the list of importers
-		const {
-			options: { admin_url: adminUrl },
-		} = site;
-
-		const otherImportersCard = (
-			<CompactCard
-				key="other-importers-card"
-				href={ adminUrl + 'import.php' }
-				target="_blank"
-				rel="noopener noreferrer"
-			>
-				{ this.props.translate( 'Choose from full list' ) }
-			</CompactCard>
+		return (
+			<>
+				{ importerElements }
+				<CompactCard
+					href={ site.options.admin_url + 'import.php' }
```

Comment:
We were previously destructuring those out from the `site` object. Any chance `site` or `site.options` would be `undefined` / `null`?

Reply:
There's very little chance one of these would be `null`. The `site` object is coming from the `getSelectedSite()` selector, the route wouldn't get rendered if the site in the route wasn't valid, and `site.options` is always present in the REST response.

Besides that, the original destructuring didn't check for `null` values either.
```js
const { options: { admin_url: adminUrl } } = site;
```
and
```js
const adminUrl = site.options.admin_url;
```
are exactly the same thing. Babel's destructuring transform [will transpile](https://babeljs.io/repl#?browsers=firefox%20%3E%2030%0A&build=&builtIns=false&spec=false&loose=false&code_lz=GYVwdgxgLglg9mABADwBQGcZQKYEpEDeAUIohAulIYnAA6wUBc1AhgCYC2MYA-iAE4AbZuy5gAqkMQBfGYgC8iTDiLSgA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=false&presets=env%2Creact&prettier=false&targets=&version=7.12.12&externalPlugins=) the first form to the second.